lib/media/track/localmediatrack.js

```diff
@@ -189,12 +189,19 @@ function mixinLocalMediaTrack(AudioOrVideoTrack) {
       // stopped, this should fire a "stopped" event.
       this._stop();
 
-      return this._trackSender.setMediaStreamTrack(mediaStreamTrack).catch(error => {
-        this._log.warn('setMediaStreamTrack failed:', { error, mediaStreamTrack });
-      }).then(() => {
-        this._initialize();
-        this._getAllAttachedElements().forEach(el => this._attach(el));
-      });
+      // If there's an unprocessedTrack, this means RTCRtpSender has
+      // the processedTrack already set, we don't want to replace that.
+      return (this._unprocessedTrack ? (() => {
+        this._unprocessedTrack = mediaStreamTrack;
+        return Promise.resolve();
+      })() :
+        this._trackSender.setMediaStreamTrack(mediaStreamTrack).catch(error => {
+          this._log.warn('setMediaStreamTrack failed:', { error, mediaStreamTrack });
+        }))
+        .then(() => {
+          this._initialize();
+          this._getAllAttachedElements().forEach(el => this._attach(el));
+        });
```

Comment:
This can be:
```js
return this._unprocessedTrack ? Promise.resolve().then(() => {
  this._unprocessedTrack = mediaStreamTrack;
}) : this._trackSender.setMediaStreamTrack(mediaStreamTrack).catch(error => {
  this._log.warn('setMediaStreamTrack failed:', { error, mediaStreamTrack });
})).then(() => {
  this._initialize();
  this._getAllAttachedElements().forEach(el => this._attach(el));
});
```